File: Practical_6/s21525600.php
<?php

if ($argc != 3) {
    echo "Usage: php s99999999.php7 <input_file> <word>\n";
    exit(1);
}

$inputFile = $argv[1];
$searchWord = strtolower($argv[2]);

if (!file_exists($inputFile) || !is_readable($inputFile)) {
    echo "The file does not exist or is not readable\n";
    exit(1);
}

$fileContent = file_get_contents($inputFile);
$formattedContent = strtolower($fileContent);
$formattedContent = preg_replace("/[^a-z]/", "", $formattedContent);

$reversedContent = strrev($formattedContent);
echo "{$reversedContent}\n";
$matches = substr_count($reversedContent, $searchWord);

echo "<html>\n<body>\n";
echo "<p>Matches: {$matches}</p>\n";
echo "</body>\n</html>";

?>

File: Practical_6/s21525600.pl
#!/usr/bin/perl

use strict;
use warnings;

my $input_file = $ARGV[0];
my $search_word = lc($ARGV[1]);

open my $fh, '<', $input_file or die "Cannot open file: $!\n";
my $file_content = do { local $/; <$fh> };
close $fh;

$file_content =~ s/[^a-zA-Z]//g;
$file_content = lc($file_content);

my $reversed_content = reverse $file_content;
my $count = () = $reversed_content =~ /$search_word/g;

print "Matches: $count\n";
